src/database/src/js/postGresOperations.js

```diff
@@ -0,0 +1,214 @@
+ *
+ */
+fluid.postgresdb.initConnection = function (that) {
+    that.pool = new pg.Pool({
+        user:       that.options.user,
+        password:   that.options.password,
+        host:       that.options.host,
+        database:   that.options.database,
+        port:       that.options.port
+    });
+};
+
+/**
+ * Convenience function to surface node-postgres `query()` function and log any
+ * errors.
+ *
+ * @param {Object} pool - node-postgres Pool instance to use for querying.
+ * @param {String} queryString - SQL query.
+ * @return {Promise} A promise whose value is the results of the query.
+ */
+fluid.postgresdb.query = function (pool, queryString) {
+    var promise = pool.query(queryString);
+    promise.then(null, function (error) {
+        fluid.log(error.message);
+    });
+    return promise;
+};
+
+/**
+ * Utility to execute an array of database queries in bulk.  This can be used to
+ * create or upgrade a set of tables in batch, or bulk load a set of records.
+ * Although not required, the array can express a logical sequence of queries,
+ * where a single query's position in the array indicates that it must come
+ * before a subsequent query, or after a previous query.
+ *
+ * @param {Object} that - Postgres request object.
+ * @param {Array} queryArray - An array of SQL statements.
+ * @return {Promise} A promise whose values are the results of running the
+ *                   sequence of queries in the `queryArray`.
+ */
+fluid.postgresdb.bulkQuery = function (that, queryArray) {
+    var querySequence = [];
+    fluid.each(queryArray, function (aQuery) {
+        querySequence.push(that.query(aQuery));
+    });
+    return fluid.promise.sequence(querySequence).then(
+        null,
+        function (error) {
+            fluid.log(error.message);
+        }
+    );
+};
+
+/**
+ * Utility to insert a set of records into a table given an array of JSON
+ * objects.
+ *
+ * The structure of the JSON must match the structure of the table.  That is,
+ * the names of the fields must match the column names, and the field values
+ * must match the value types declared for the table.  In addition, where the
+ * table column requires a non-null value and no default is specified, there
+ * must be a corresponding name/value pair in the JSON object.  If the table
+ * column value is allowed to be null or has a default, the corresponding JSON
+ * name/value pair can be omitted.
+ *
+ * @param {Object} that - Postgres request object.
+ * @param {Sting} tableName - Name of table to insert into.
+ * @param {Array} jsonArray - An array of JSON objects to load.
+ * @return {Promise} whose value is an array of successful INSERT results.
+ */
+fluid.postgresdb.loadFromJSON = function (that, tableName, jsonArray) {
+    var insertions = [];
+    fluid.each(jsonArray, function (aRecord) {
+        var columns = fluid.postgresdb.stringifyJoinKeys(Object.keys(aRecord));
```

Comment:
As the node-postgres docs explain, raw string-bashing for assembling queries like this is highly dangerous. You should apply a parameterised query as documented here: https://node-postgres.com/features/queries#parameterized-query
As @cindyli has suggested, the various "stringify" functions below are problematic and should be removed